site/gdocs/SDGTableOfContents.tsx

```diff
@@ -0,0 +1,87 @@
+import { TocHeading } from "../../clientUtils/owidTypes.js"
+import { faArrowDown } from "@fortawesome/free-solid-svg-icons/faArrowDown"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome/index.js"
+import { faPlus } from "@fortawesome/free-solid-svg-icons/faPlus"
+import { faMinus } from "@fortawesome/free-solid-svg-icons/faMinus"
+import classNames from "classnames"
+import AnimateHeight from "react-animate-height"
+
+const VERTICAL_TAB_CHAR = "\u000b"
+
+export default function SDGTableOfContents({ toc }: { toc: TocHeading[] }) {
+    const [height, setHeight] = useState<"auto" | 0>(0)
+    const [isOpen, setIsOpen] = useState(false)
+
+    const toggleIsOpen = () => {
+        setHeight(height === 0 ? "auto" : 0)
+    }
+
+    const tocHeadingsWithSupertitle = toc.map((heading) => {
+        const [beforeSeparator, afterSeparator] =
+            heading.text.split(VERTICAL_TAB_CHAR)
+
+        return {
+            ...heading,
+            supertitle: afterSeparator ? beforeSeparator : undefined,
+            title: afterSeparator || beforeSeparator,
+        }
+    })
+
+    return (
+        <nav className={classNames("sdg-toc", { open: isOpen })}>
+            <div
+                onClick={toggleIsOpen}
+                className="sdg-toc-toggle"
+                data-track-note="sdg-toc-toggle"
+            >
+                <div>Index</div>
+                <div>
+                    <FontAwesomeIcon icon={isOpen ? faMinus : faPlus} />
+                </div>
+            </div>
+            <AnimateHeight
+                className="sdg-toc-content"
+                height={height}
+                onHeightAnimationStart={(newHeight) => {
+                    if (newHeight != 0) setIsOpen(true)
+                }}
+                onHeightAnimationEnd={(newHeight) => {
+                    if (newHeight === 0) setIsOpen(false)
+                }}
+                animateOpacity
+            >
+                <ul>
+                    {tocHeadingsWithSupertitle.map(
+                        (
+                            { title, supertitle, isSubheading, slug },
+                            i: number
+                        ) => (
+                            <li
+                                key={i}
+                                className={
+                                    isSubheading ? "subsection" : "section"
+                                }
+                            >
+                                <a
+                                    href={`#${slug}`}
+                                    data-track-note="sdg-toc-link"
+                                >
+                                    {supertitle ? (
+                                        <span className="supertitle">
+                                            {supertitle}
+                                        </span>
+                                    ) : null}
+                                    {title}
```

Comment:
I think we need a space in here somewhere

![image](https://user-images.githubusercontent.com/11844404/199259338-ea7215b3-7994-4a28-aed2-9dd0f05b66f5.png)

Reply:
hum .. did you do anything specific to render it that way? There is a actually a space, in an ::after pseudo-element in css:

https://github.com/owid/owid-grapher/blob/9591c5f078047bbdd2602e4351aaac0018ce44f2/site/gdocs/SDGTableOfContents.scss#L63

As a side note, I don't quite recall why I did it this way, maybe to keep the JS code more readable, maybe to avoid an unnecessary space in section headings without adding conditional logic. Both of those are marginal concerns here, so let me know if you have a strong reason not to do it this way.